### ISSUE TITLE
Not syncing VTODO with Tasks application

### ISSUE THREAD

**Issue by MohammadFakhraee** [bitfireAT/ical4android#69]
Hi there. I'm currently facing a problem in which the tasks are not being synced with server and local application (Tasks). The platforms I'm using are:
1. NextCloud server (I've tested The Good Cloud & Qloud providers),
2. DAVx5 for synchronizing tasks,
3. Tasks application as a client to see the tasks.
All the synchronization works fine when I log in to my account for the first time using DAVx5. It syncs the task data, and Tasks would show them with the latest changes.
But when I make any changes in the web service, then clicking on sync button inside DAVx5, and navigating to Tasks application, nothing is updated. The problem persists vise versa. Like, if I make any changes using Tasks and clicking on DAVx5's sync button , nothing would change in web service.

I tried changing service provider but the problem was still existing, so it is NOT the server.
I tried using Jtx Board to see if the problem exists. Surprisingly, the tasks were updating after any changes.
So I am not completely sure if it's Tasks application's bug or DAVx5's. But I'm sure the synchronization between these two, are working fine for the first time and  are not working perfectly for the afterward changes.

**Comment by patrickunterwegs** [comment]
Hello @MohammadFakhraee ,
Are you aware of the setting to change the tasks provider?

![Screenshot_20221226_101136_DAVx.jpg](https://user-images.githubusercontent.com/72232737/209530063-4689aa4d-31af-4e4b-8bb5-a14dc2528b72.jpg)

DAVx5 can synchronize with one specific tasks app (not with both at once). Maybe the setting was updated to jtx Board after you installed it and now the sync for the other tasks app is blocked.

**Comment by MohammadFakhraee** [comment]
Yes I did select the Tasks app in Integration section and allowed the permission of the Tasks, and clicked the sync to update the provider but it did not updated.

**Comment by MohammadFakhraee** [comment]
I even found same [issue reported to Tasks app](https://github.com/tasks/tasks/issues/2078)

**Comment by MohammadFakhraee** [comment]
I'm not sure what is the related explanation but the problem was because of the AGP version. I was using android dolphin so I couldn't build Tasks project with `AGP 7.4.0-rc02`. So I downgraded to `gradle 7.4-all` and `AGP 7.3.1` so I could run it. But I downloaded Android Studio Canary now and could build the project with `AGP 7.4.0-rc02` successfully. And the synchronization is working perfectly.
@patrickunterwegs for your answer dude.
